code/modules/projectiles/projectile.dm

```diff
@@ -67,10 +67,26 @@
 	qdel(src)
 
 /obj/item/projectile/proc/on_hit(atom/target, blocked = 0, hit_zone)
+	var/turf/target_loca = get_turf(target)
 	if(!isliving(target))
 		return 0
 	var/mob/living/L = target
 	if(blocked < 100) // not completely blocked
+		if(damage && L.blood_volume && damage_type == BRUTE)
+			var/splatter_dir = dir
+			if(starting)
+				splatter_dir = get_dir(starting, target_loca)
+			if(isalien(L))
+				new /obj/effect/overlay/temp/dir_setting/bloodsplatter/xenosplatter(target_loca, splatter_dir)
+			else
+				var/blood_color = "#C80000"
+				if(ishuman(target))
+					var/mob/living/carbon/human/H = target
+					blood_color = H.species.blood_color
+				new /obj/effect/overlay/temp/dir_setting/bloodsplatter(target_loca, splatter_dir, blood_color)
+			if(prob(33))
+				L.add_splatter_floor(target_loca)

```

Comment:
Can check if target_loca is a wall/fullwindow/etc. here and decide whether to add_splatter_floor or whatever the other proc is